route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -41,190 +41,238 @@ namespace route_following_plugin
-            {
-                //calculate required distance for lane change
-                double longl_travel_dist = (target_speed*(LANE_CHANGE_TIME_MAX + buffer_lanechange_time_));
-                double lane_change_start_dist;                             
-                
-                if(wm_->routeTrackPos(shortest_path[last_lanelet_index + 1].centerline2d().back()).downtrack >= route_length){
-                    lane_change_start_dist = route_length - longl_travel_dist;
-                    end_dist = route_length;
-
-                }
-                else{
-                    lane_change_start_dist = wm_->routeTrackPos(shortest_path[last_lanelet_index + 1].centerline2d().back()).downtrack - longl_travel_dist;
-                    end_dist  = wm_->routeTrackPos(shortest_path[last_lanelet_index + 1].centerline2d().back()).downtrack;
-                }               
-
-                lane_change_start_dist = current_progress;
-                double starting_lanelet_id = shortest_path[last_lanelet_index].id();
-                double ending_lanelet_id = shortest_path[last_lanelet_index+1].id();
-                resp.new_plan.maneuvers.push_back(
-                composeLaneChangeManeuverMessage(lane_change_start_dist, end_dist, speed_progress, target_speed, 
-                                    starting_lanelet_id, ending_lanelet_id,
-                                    time_progress));
-                
-                ++last_lanelet_index;
-            }
-            else
-            {
-                resp.new_plan.maneuvers.push_back(
-                composeManeuverMessage(current_progress, end_dist,  
-                                    speed_progress, target_speed, 
-                                    shortest_path[last_lanelet_index].id(), time_progress));
-                
-            }
-            current_progress += dist_diff;
-            //time_progress = resp.new_plan.maneuvers.back().lane_following_maneuver.end_time;
-            speed_progress = target_speed;
-            
-
-            if(current_progress >= total_maneuver_length || last_lanelet_index == shortest_path.size() - 1)
-            {
-                break;
-            }
-
-            ++last_lanelet_index;
+            prev_mvr_end_speed = target_speed_in_lanelet;
+        }
+        //add lane follow as last maneuver
+        if(maneuvers.size() < route_shortest_path.size()){
+            double target_speed_in_lanelet = findSpeedLimit(route_shortest_path.back());
+            double start_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().front()).downtrack;
+            double end_dist = wm_->routeTrackPos(route_shortest_path.back().centerline2d().back()).downtrack; 
+            maneuvers.push_back(composeLaneFollowingManeuverMessage(start_dist, end_dist, prev_mvr_end_speed, target_speed_in_lanelet, route_shortest_path.back().id(),ros::Time::now()));
```

Comment:
there us a ros::Time now field here as well which might need to change based on your discussion with Kyle

Reply:
The ros::Time now here is just a place holder, the time is updated without using this info when writing to response.
This happens in updateTimeProgress